articles/cookie-token-auth/index.md

```diff
@@ -0,0 +1,135 @@
+
+Here is a logical flow of the cookie-based authentication process:
+
+1. The client sends a login request with credentials to the backend server via the client interface.
+
+2. The server then validates the credentials. If the login is successful, the web server will create a session in the database and include a `Set-Cookie` header on the response containing a unique ID in the cookie object.
+
+3. The browser saves the cookie locally. As long as the user stays logged in, the client needs to send the cookie in all the subsequent requests to the server. The server then compares the session ID stored in the cookie against the one in the database to verify the validity.
+   
+4. During the logout operation, the server will make the cookie expire by deleting it from the database.
+
+### Advantages of cookie-based authentication
+1. By using cookies in authentication makes your application stateful. This will be efficient in tracking and personalizing the state of a user.
+
+2. Cookies are small in size, making them efficient to store on the client-side.
+
+3. Cookies can be "HTTP-only" making them impossible to read on the client-side. This improves protection against any Cross-site scripting (XSS) attacks.
+   
+4. Cookies will be added to the request automatically, so the developer will not have to implement them manually and therefore less code.
+
+### Limitations of cookie-based authentication
+1. It is vulnerable to [Cross-site request forgery attack](https://www.mozilla.developer.org/en-US/docs/Glossary/CSRF). It often needs other security measures such as [CSRF tokens](https://www.portswigger.net/websecurity/csrf/tokens) for protection.
+   
+2. You need to store the session data in a database or keep it in memory on the server. This makes it less scalable and increases overhead when the site is having many users.
+
+3. Cookies normally work on a single domain. For example, it is impossible to read or send a cookie from a domain like jabs.com to a boo.com domain. This is an issue when the API service is from different domains for mobile and web platforms. 
+
+4. The client needs to send a cookie on every request, even with the URLs that do not need authentication for access.
+
+### Understanding Token-based authentication
+In token-based authentication, we store the user's state on the client. JSON Web Token(JWT) is an open standard [(RFC 7519)](https://tools.ietf.org/html/rfc7519) that defines a way of securely transmitting information between a client and a server as a JSON object. I will use tokens and JWT terms interchangeably in the article.
+
+The anatomy of a JWT token comprises three parts separated by dots(.). The three parts include the JWT header, the JWT payload, and its signature, respectively (`header.payload.signature`).
```

Comment:
Please provide an example of a JWT token. Also, instruct that readers can use the jwt.io to parse the information before moving on to the examples.